fix infinite alert when game is over

masquel opened the pull request:
Fix issue #1

Commit:
fix infinite alert when game is over

Commit:
reset score if game is over

PezCoder commented on package-lock.json:
```diff
@@ -0,0 +1,3 @@
+{
```
The game doesn't not using any external package, so we mightn't need this file.

masquel replied:
Yeah, sure. This was my habit to run `npm install` after cloning repo, so this file shouldn't be in repo.

PezCoder commented on snake.js:
```diff
@@ -287,9 +284,15 @@
+      gameOver();
+    }
+  }
+  function gameOver(){
```
Good idea for resetting the state, but there are more states in the game to keep track of & not just lives & score, Like the direction of the snake, no of snake blocks.

I think we can keep the reloading logic for now, but an `isGameOver`  boolean check with `lives === 0` should suffice.

masquel replied:
You can move all game data you need to reset after game is over to `gameData` object for example, for easy reset in future.

PezCoder replied:
Yeah, good idea.

PezCoder commented on snake.js:
```diff
@@ -287,9 +284,15 @@
     }
-
+    if(lives === 0){
+      gameOver();
```
Any specific reason for putting this inside `updateHead()` ? 
I think the purpose of updateHead should be to only update the positions of head & not anything else, It'll make it easier to track bugs later on & the code will be easier to understand.

The code related to game state should be belong in the main game loop i.e `draw()` function

PezCoder reviewed (changes requested):
Thank you for the pr 👏🏼
I have a few suggestions.

Commit:
remove package-lock.json, clean up, add restart game func, add more example of reset data

masquel commented:
@PezCoder I've moved check lives to `draw` function, added `resetGame` function in which you should reset all game data you need. All data i've reseted in this function is just for example. It's an idea how to solve your problem. Also i've remove package-lock.json
